src/java.base/unix/classes/sun/nio/fs/UnixSecureDirectoryStream.java

```diff
@@ -414,15 +414,24 @@ public void setPermissions(Set<PosixFilePermission> perms)
                 if (!ds.isOpen())
                     throw new ClosedDirectoryStreamException();
 
-                int fd = (file == null) ? dfd : open();
-                try {
-                    fchmod(fd, UnixFileModeAttribute.toUnixMode(perms));
-                } catch (UnixException x) {
-                    x.rethrowAsIOException(file);
-                } finally {
-                    if (file != null && fd >= 0)
-                        UnixNativeDispatcher.close(fd, e-> null);
+                if (file == null)
+                    fchmod(dfd, UnixFileModeAttribute.toUnixMode(perms));
+                else if (followLinks)
+                    fchmodat(dfd, file, UnixFileModeAttribute.toUnixMode(perms), 0);
+                else if (fchmodatNoFollowSupported())
+                    fchmodat(dfd, file, UnixFileModeAttribute.toUnixMode(perms),
+                             AT_SYMLINK_NOFOLLOW);
```

Comment:
The 4 cases look okay. If you adds`int mode = UnixFileModeAttribute.toUnixMode(perms)` then it would avoid the repeated code and avoid the line splits, just makes it easier to read.

Reply:
So changed in https://github.com/openjdk/jdk/pull/25534/commits/66de3e524567deb8718b70c394c8877745de16fe.